pkg/reloader/reloader.go

```diff
@@ -692,16 +693,24 @@ func RuntimeInfoURLFromBase(u *url.URL) *url.URL {
 
 var envRe = regexp.MustCompile(`\$\(([a-zA-Z_0-9]+)\)`)
 
-func expandEnv(b []byte) (r []byte, err error) {
+func expandEnv(logger log.Logger, b []byte, suppressEnvironmentVariablesExpansionErrors bool, configApplyErrorsPtr *prometheus.Counter) (r []byte, err error) {
```

Comment:
rather than passing the additional parameters, why not make it a `*Reloader`  function?